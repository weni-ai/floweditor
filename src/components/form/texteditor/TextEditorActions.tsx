import * as React from 'react';
import { StringEntry } from 'store/nodeEditor';
import { applyVueInReact } from 'veaury';

// @ts-ignore
import Unnnic from '@weni/unnnic-system';

import styles from './TextEditorActions.module.scss';

const UnnnicButton = applyVueInReact(Unnnic.unnnicButton);
const UnnnicEmojiPicker = applyVueInReact(Unnnic.unnnicEmojiPicker);

export interface TextEditorActionProps {
  entry: StringEntry;
  maxLength: number;
  onAddEmoji: (emoji: any) => void;
  onFormat: (c: string) => void;
}

export interface TextEditorState {
  showEmojiPicker: boolean;
}

export default class TextEditorActions extends React.Component<
  TextEditorActionProps,
  TextEditorState
> {
  constructor(props: TextEditorActionProps) {
    super(props);

    this.state = {
      showEmojiPicker: false,
    };
  }

  private toggleEmojiPicker(): void {
    this.setState(prevState => ({
      showEmojiPicker: !prevState.showEmojiPicker,
    }));
  }

  private addEmoji(emoji: any) {
    this.props.onAddEmoji(emoji);
    this.toggleEmojiPicker();
  }

  render() {
    return (
      <div className={styles.actions}>
        <div className={styles.emoji_picker_wrapper}>
          <UnnnicButton
            data-testid="emoji-button"
            iconCenter="sentiment_satisfied"
            size="small"
            type="tertiary"
            onClick={() => this.toggleEmojiPicker()}
          />
          {this.state.showEmojiPicker && (
            <div className={styles.emoji_picker}>
              <UnnnicEmojiPicker
                data-testid="emoji-picker"
                onEmojiSelected={(e: any) => this.addEmoji(e)}
                onClickOutside={() => this.toggleEmojiPicker()}
              />
            </div>
          )}
        </div>

        <UnnnicButton
          data-testid="bold-button"
          iconCenter="format_bold"
          size="small"
          type="tertiary"
          onClick={() => this.props.onFormat('*')}
        />

        <UnnnicButton
          data-testid="italic-button"
          iconCenter="format_italic"
          size="small"
          type="tertiary"
          onClick={() => this.props.onFormat('_')}
        />

        <UnnnicButton
          data-testid="code-button"
          iconCenter="code"
          size="small"
          type="tertiary"
          onClick={() => this.props.onFormat('```')}
        />

        <UnnnicButton
          data-testid="strike-button"
          iconCenter="format_strikethrough"
          size="small"
          type="tertiary"
          onClick={() => this.props.onFormat('~')}
        />

        <span className={styles.counter}>
          {this.props.entry.value.length} / {this.props.maxLength}
        </span>
      </div>
    );
  }
}
