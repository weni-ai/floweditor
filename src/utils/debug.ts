import { FlowStoreProps } from 'components/flow/Flow';
import { DebugState } from 'store/editor';

import mutate from 'immutability-helper';

/* istanbul ignore next -- @preserve */
export default class Debug {
  private props: FlowStoreProps;
  private state: DebugState;

  constructor(props: FlowStoreProps, initial: DebugState) {
    this.props = props;
    this.state = initial || { showUUIDs: false };
  }

  public showUUIDs(): DebugState {
    const updated = mutate(this.state, { $merge: { showUUIDs: true } });
    this.props.mergeEditorState({ debug: updated });
    return updated;
  }
}
