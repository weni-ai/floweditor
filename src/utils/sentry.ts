import * as Sentry from '@sentry/react';

const initSentry = (dsn: string) => {
  Sentry.init({
    dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.httpClientIntegration({
        failedRequestStatusCodes: [[400, 499], [500, 599]],
      }),
      Sentry.replayIntegration(),
      Sentry.replayCanvasIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
};

export default initSentry;
