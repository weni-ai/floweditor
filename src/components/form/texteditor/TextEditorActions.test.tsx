import TextEditorActions, {
  TextEditorActionProps,
} from 'components/form/texteditor/TextEditorActions';
import * as React from 'react';
import { render, fireEvent, fireCustomEvent } from 'test/utils';

const baseProps: TextEditorActionProps = {
  entry: { value: '' },
  maxLength: 100,
  onAddEmoji: vi.fn(),
  onFormat: vi.fn(),
};

describe(TextEditorActions.name, () => {
  it('renders', () => {
    const { baseElement } = render(<TextEditorActions {...baseProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should toggle emoji picker', () => {
    const { baseElement, getByTestId } = render(
      <TextEditorActions {...baseProps} />,
    );

    const emojiButton = getByTestId('emoji-button');
    expect(baseElement).toMatchSnapshot();
    fireEvent.click(emojiButton);
    expect(baseElement).toMatchSnapshot();
  });

  it('should emit add emoji', () => {
    const { getByTestId } = render(<TextEditorActions {...baseProps} />);

    const emojiButton = getByTestId('emoji-button');
    fireEvent.click(emojiButton);

    expect(baseProps.onAddEmoji).not.toHaveBeenCalled();
    const emojiPicker = getByTestId('emoji-picker');

    fireCustomEvent(emojiPicker, 'emoji-selected', { emoji: '👍' });
    expect(baseProps.onAddEmoji).toHaveBeenCalled();
  });

  it('should close emoji picker on click outside', () => {
    const { baseElement, getByTestId } = render(
      <TextEditorActions {...baseProps} />,
    );

    const emojiButton = getByTestId('emoji-button');
    fireEvent.click(emojiButton);
    expect(baseElement).toMatchSnapshot();

    const emojiPicker = getByTestId('emoji-picker');
    fireCustomEvent(emojiPicker, 'click-outside', true);
    expect(baseElement).toMatchSnapshot();
  });

  it('should emit formats', () => {
    const { getByTestId } = render(<TextEditorActions {...baseProps} />);

    const boldButton = getByTestId('bold-button');
    const italicButton = getByTestId('italic-button');
    const codeButton = getByTestId('code-button');
    const strikeButton = getByTestId('strike-button');

    expect(baseProps.onFormat).not.toHaveBeenCalled();
    fireEvent.click(boldButton);
    expect(baseProps.onFormat).toHaveBeenCalledWith('*');

    fireEvent.click(italicButton);
    expect(baseProps.onFormat).toHaveBeenCalledWith('_');

    fireEvent.click(codeButton);
    expect(baseProps.onFormat).toHaveBeenCalledWith('```');

    fireEvent.click(strikeButton);
    expect(baseProps.onFormat).toHaveBeenCalledWith('~');
  });
});
