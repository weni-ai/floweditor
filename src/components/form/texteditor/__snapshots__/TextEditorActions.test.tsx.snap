// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextEditorActions > renders 1`] = `
<body>
  <div>
    <div
      class="_actions_e669ba"
    >
      <div
        class="_emoji_picker_wrapper_e669ba"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <button
            class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
            data-testid="emoji-button"
            data-v-4cec5e43=""
          >
            <!---->
            <!---->
            <span
              class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
              data-v-4cec5e43=""
              data-v-930380cc=""
            >
              sentiment_satisfied
            </span>
            <span
              class="unnnic-button__label"
              data-v-4cec5e43=""
            >
              
              
               
            </span>
            <!---->
          </button>
        </div>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="bold-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_bold
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="italic-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_italic
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="code-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            code
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="strike-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_strikethrough
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <span
        class="_counter_e669ba"
      >
        0
         / 
        100
      </span>
    </div>
  </div>
</body>
`;

exports[`TextEditorActions > should close emoji picker on click outside 1`] = `
<body>
  <div>
    <div
      class="_actions_e669ba"
    >
      <div
        class="_emoji_picker_wrapper_e669ba"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <button
            class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
            data-testid="emoji-button"
            data-v-4cec5e43=""
          >
            <!---->
            <!---->
            <span
              class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
              data-v-4cec5e43=""
              data-v-930380cc=""
            >
              sentiment_satisfied
            </span>
            <span
              class="unnnic-button__label"
              data-v-4cec5e43=""
            >
              
              
               
            </span>
            <!---->
          </button>
        </div>
        <div
          class="_emoji_picker_e669ba"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <div
              class="emoji-picker"
              data-testid="emoji-picker"
              data-v-67fb0974=""
            >
              <em-emoji-picker />
            </div>
          </div>
        </div>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="bold-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_bold
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="italic-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_italic
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="code-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            code
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="strike-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_strikethrough
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <span
        class="_counter_e669ba"
      >
        0
         / 
        100
      </span>
    </div>
  </div>
</body>
`;

exports[`TextEditorActions > should close emoji picker on click outside 2`] = `
<body>
  <div>
    <div
      class="_actions_e669ba"
    >
      <div
        class="_emoji_picker_wrapper_e669ba"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <button
            class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
            data-testid="emoji-button"
            data-v-4cec5e43=""
          >
            <!---->
            <!---->
            <span
              class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
              data-v-4cec5e43=""
              data-v-930380cc=""
            >
              sentiment_satisfied
            </span>
            <span
              class="unnnic-button__label"
              data-v-4cec5e43=""
            >
              
              
               
            </span>
            <!---->
          </button>
        </div>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="bold-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_bold
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="italic-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_italic
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="code-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            code
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="strike-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_strikethrough
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <span
        class="_counter_e669ba"
      >
        0
         / 
        100
      </span>
    </div>
  </div>
</body>
`;

exports[`TextEditorActions > should toggle emoji picker 1`] = `
<body>
  <div>
    <div
      class="_actions_e669ba"
    >
      <div
        class="_emoji_picker_wrapper_e669ba"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <button
            class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
            data-testid="emoji-button"
            data-v-4cec5e43=""
          >
            <!---->
            <!---->
            <span
              class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
              data-v-4cec5e43=""
              data-v-930380cc=""
            >
              sentiment_satisfied
            </span>
            <span
              class="unnnic-button__label"
              data-v-4cec5e43=""
            >
              
              
               
            </span>
            <!---->
          </button>
        </div>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="bold-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_bold
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="italic-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_italic
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="code-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            code
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="strike-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_strikethrough
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <span
        class="_counter_e669ba"
      >
        0
         / 
        100
      </span>
    </div>
  </div>
</body>
`;

exports[`TextEditorActions > should toggle emoji picker 2`] = `
<body>
  <div>
    <div
      class="_actions_e669ba"
    >
      <div
        class="_emoji_picker_wrapper_e669ba"
      >
        <div
          data-use-vue-component-wrap=""
          data-v-app=""
          style="all: unset;"
        >
          <button
            class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
            data-testid="emoji-button"
            data-v-4cec5e43=""
          >
            <!---->
            <!---->
            <span
              class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
              data-v-4cec5e43=""
              data-v-930380cc=""
            >
              sentiment_satisfied
            </span>
            <span
              class="unnnic-button__label"
              data-v-4cec5e43=""
            >
              
              
               
            </span>
            <!---->
          </button>
        </div>
        <div
          class="_emoji_picker_e669ba"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="all: unset;"
          >
            <div
              class="emoji-picker"
              data-testid="emoji-picker"
              data-v-67fb0974=""
            >
              <em-emoji-picker />
            </div>
          </div>
        </div>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="bold-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_bold
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="italic-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_italic
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="code-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            code
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <div
        data-use-vue-component-wrap=""
        data-v-app=""
        style="all: unset;"
      >
        <button
          class="unnnic-button unnnic-button--size-small unnnic-button--tertiary unnnic-button--icon-on-center"
          data-testid="strike-button"
          data-v-4cec5e43=""
        >
          <!---->
          <!---->
          <span
            class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--sm"
            data-v-4cec5e43=""
            data-v-930380cc=""
          >
            format_strikethrough
          </span>
          <span
            class="unnnic-button__label"
            data-v-4cec5e43=""
          >
            
            
             
          </span>
          <!---->
        </button>
      </div>
      <span
        class="_counter_e669ba"
      >
        0
         / 
        100
      </span>
    </div>
  </div>
</body>
`;
